### ISSUE TITLE
Default flags do not work for Intel on Windows

### ISSUE THREAD

**Issue by everythingfunctional** [fortran-lang/fpm#383]
The default flags do not work for `ifort` on Windows. I get:

```
ifort: command line error: Unrecognized keyword 'all:noarg_temp_created' for option '/check'
```

**Comment by everythingfunctional** [comment]
Follow up, running with the `--release` option *kinda works*, in that it will compile stuff, but it complains about ignoring unknown options:

```
ifort: command line warning #10159: invalid argument for option '/fp'
ifort: command line warning #10161: unrecognized source type 'precise'; object file assumed
ifort: command line warning #10006: ignoring unknown option '/pc'
ifort: command line warning #10161: unrecognized source type '64'; object file assumed
ifort: command line warning #10161: unrecognized source type 'all'; object file assumed
ifort: command line warning #10155: ignoring option '/error-limit'; argument required
ifort: command line warning #10161: unrecognized source type '1'; object file assumed
ifort: command line warning #10161: unrecognized source type 'threaded'; object file assumed
ifort: command line warning #10155: ignoring option '/assume'; argument required
ifort: command line warning #10161: unrecognized source type 'byterecl'; object file assumed
ifort: command line warning #10155: ignoring option '/module'; argument required
ifort: command line warning #10161: unrecognized source type 'build\ifort_release\hello_fpm'; object file assumed
```

**Comment by awvwgk** [comment]
Intel on Windows is using MSVS compatible syntax rather than GCC compatible syntax for the option and is usually considered a “different” compiler than Intel on Unix (meson has different identifiers and different defaults for them).

**Comment by urbanjost** [comment]
So the most obvious question is should there just be another select case added with a different name like "ifort-msvs" or should the compiler name "ifort" work and something like the OS_TYPE() routine select Microsoft or Unix-Like switches  (or is there some way to allow the Microsoft version to accept the Unix syntax?

**Comment by everythingfunctional** [comment]
I think we want the name "ifort" to still work, and then use the `OS_TYPE` to select the set of switches for the current OS.

Note that it's not just the syntax that's different. The Intel compiler legitimately has a different set of available options for different OSes. In fact, with this fix we could add `-coarray=single` back to the default set for Linux, while leaving it off for Mac.

**Comment by urbanjost** [comment]
That's a good justification for going the OS_TYPE() route. Seems cleaner. On the other side of that, might it be confusing to use the same name and get different results on different platforms?   I think I like the OS_TYPE() approach the best, but a little hesitant;  but do not like taking a "good" default off the Linux options because not all platforms support it either.

**Comment by everythingfunctional** [comment]
I'd argue that Intel are the ones who gave it the same name and get different results on different platforms. I think as far fpm is concerned, we want things to "just work" for as many people on as many platforms as possible. Somebody who was able to do `fpm run --compiler ifort` on one platform should be able to run the exact same command on another and expect it to work.

**Comment by everythingfunctional** [comment]
Fixed by #431
